task3_snkrs/spiders/orsay_spider.py

```diff
@@ -0,0 +1,112 @@
+
+    def parse_product(self, response):
+        item = Item()
+        item['item_id'] = self.get_item_id(response)
+        item['name'] = self.get_name(response)
+        item['description'] = self.get_description(response)
+        item['skus'] = self.get_skus(response)
+        item['image_urls'] = self.get_image_urls(response)
+        item['meta'] = self.color_requests(item, response)
+        color_requests = item['meta']
+
+        return color_requests.pop() if color_requests else item
+
+    def parse_color(self, response):
+        item = response.meta['item']
+        item['skus'] += self.get_skus(response)
+        item['image_urls'] += self.get_image_urls(response)
+        color_requests = item['meta']
+
+        return color_requests.pop() if color_requests else item
+
+    def color_requests(self, item, response):
+        color_urls = response.css('ul[class="swatches color"] li[class="selectable"] a::attr(href)').getall()
+
+        return [scrapy.Request(url, callback=self.parse_color, meta={'item': item})
+                for url in color_urls]
+
+    def get_item_id(self, response):
+        return response.css('div.product-tile::attr(data-itemid)').get()
+
+    def get_name(self, response):
+        return response.css('.product-name::text').get()
+
+    def get_image_urls(self, response):
+        return response.css('img.productthumbnail::attr(src)').getall()
+
+    def get_currency(self, response):
+        return response.css('.current .country-currency::text').get()
+
+    def get_availability(self, response):
+        is_available = response.css('p.in-stock-msg ::text').get()
+        return True if is_available == 'Auf Lager' else False
+
+    def get_description(self, response):
+        return self.filter_empty_elements(response.css('.with-gutter::text').getall())
+
+    def get_sizes(self, response):
+        return self.filter_empty_elements(response.css('li.selectable a.swatchanchor::text').getall())
+
+    def get_colors(self, response):
+        return self.filter_colors(response.css('.swatchanchor::attr(title)').getall())
+
+    def get_selected_color(self, response):
+        return self.filter_colors(response.css('li[class="selectable selected"] .swatchanchor::attr(title)').getall())
+
+    def get_old_prices(self, response):
+        return self.remove_commas(response.css('.price-standard::text').re('[\d,.]+'))
+
+    def get_sale_price(self, response):
+        return self.remove_commas(response.css('.price-sales::text').re('[\d,.]+'))
+
+    def remove_commas(self, elements):
+        return [e.replace(',', '') for e in elements]
+
+    def filter_empty_elements(self, data):
+        return [e.strip() for e in data if e.strip() != '']
+
+    def filter_colors(self, colors):
+        return [e.split('-')[-1].strip() for e in colors if e.strip() != '']
+
+    def get_skus(self, response):
+        sku = {}
+        for raw_sku in self.get_sizes(response) or ['size-one']:
+            sku[raw_sku] = {
```

Comment:
Let's create a method for pricing, It will return `price`, `old_prices` and currency.